### PR TITLE
Work around world age error for Duals in exp_generic

### LINKED ISSUE #42: generic_exp fails in zero
```julia
julia> ExponentialUtilities.exp_generic(0.0)
ERROR: InexactError: trunc(Int64, -Inf)
Stacktrace:
 [1] trunc at ./float.jl:703 [inlined]
 [2] ceil at ./float.jl:365 [inlined]
 [3] exp_generic(::Float64, ::Val{13}) at /Users/andreasnoack/.julia/dev/ExponentialUtilities/src/exp.jl:119
 [4] exp_generic(::Float64) at /Users/andreasnoack/.julia/dev/ExponentialUtilities/src/exp.jl:118
 [5] top-level scope at REPL[64]:1
```

### PR DESCRIPTION
**Pull request by andreasnoack**
and fix `exp_generic(0.0)`.

Closes #42 but not really #41. It's more a temporary workaround than an actual fix. I'm not sure how to fix the problem in full generality.

### CODE REVIEW

**Comment by andreasnoack** [comment]
Please hold off a bit. I need to define the zero branch differently since
```
julia> ForwardDiff.derivative(exp_generic, 0.0)
0.0
```

**Comment by andreasnoack** [comment]
Fixed it so I believe this one should be good to go.

File: test/runtests.jl [hunk 1]
```diff
@@ -14,21 +14,34 @@ using ExponentialUtilities: getH, getV, _exp!
...
-        
+
         N = randn(n, n)
         @test exp(N) ≈ exp_generic(N)
 
         exp(n) ≈ exp_generic(n)
     end
 end
 
+# The issue was only triggered if exp_generic had been called before loading
```
Comment (vtjnash) [inline test/runtests.jl:32]: This should be incorrect. What matters is the load order, while it's a bug if the call order is observable.

Comment (ChrisRackauckas) [reply test/runtests.jl:32]: how'd you find this? 😆 I think we can just straight avoid the issue for the most common case (https://github.com/SciML/ExponentialUtilities.jl/pull/45) and handle the more general thing later.

Comment (vtjnash) [reply test/runtests.jl:32]: Someone put out the bat signal 🙃

**Comment by andreasnoack** [comment]
Superseded by #45

**Pull request closed without merging**